excise/tests/test_avatax_excise.py

```diff
@@ -232,12 +251,158 @@ def test_calculate_checkout_line_total(
 
 @pytest.mark.vcr
 @pytest.mark.parametrize(
-    "with_discount, expected_net, expected_gross, voucher_amount, taxes_in_prices",
+    "with_discount, expected_net, expected_gross, taxes_in_prices",
+    [
+        (False, "30.00", "32.29", True),
+    ],
+)
+@override_settings(
+    PLUGINS=["saleor.plugins.avatax.excise.plugin.AvataxExcisePlugin"]
+)
+def test_calculate_checkout_line_total_metadata(
+    reset_sequences,  # pylint: disable=unused-argument
+    with_discount,
+    expected_net,
+    expected_gross,
+    taxes_in_prices,
+    discount_info,
+    checkout_with_item,
+    address_usa_tx,
+    address_usa,
+    site_settings,
+    shipping_zone,
+    plugin_configuration,
+):
+    plugin_configuration()
+    manager = get_plugins_manager()
+
+    checkout_with_item.shipping_address = address_usa_tx
+    checkout_with_item.shipping_method = shipping_zone.shipping_methods.get()
+    checkout_with_item.save()
+    site_settings.company_address = address_usa
+    site_settings.include_taxes_in_prices = taxes_in_prices
+    site_settings.save()
+    line = checkout_with_item.lines.first()
+    product = line.variant.product
+    product.metadata = {}
+    product.save()
+    product.product_type.save()
+    discounts = [discount_info] if with_discount else None
+
+    lines = fetch_checkout_lines(checkout_with_item)
+    checkout_info = fetch_checkout_info(
+        checkout_with_item, lines, discounts, manager
+    )
+    checkout_line_info = lines[0]
+
+    manager.calculate_checkout_line_total(
+        checkout_info,
+        lines,
+        checkout_line_info,
+        checkout_with_item.shipping_address,
+        discounts,
+    )
+
+    checkout_with_item.refresh_from_db()
+
+    target_data = [
```

Comment:
Are you sure that we need to store all this dict in metadata?

Reply:
We can make this shorter by checking if metadata was set, it has been fixed here: https://github.com/saleor/saleor-plugin-avatax-excise/pull/7/commits/584f3bed49085cd2bf5e9bd5fa00c6887d427ee1